IQSS/dataverse
# Add gdcc/nginx:unstable image and dev_nginx container for IntelliJ Payara plugin workaround

**What this PR does / why we need it**:

Adds a nginx container to the docker setup. This could be used for various proxying stuff, but it is primarily added to work around a problem with the IntelliJ Payara plugin. It doesn't allow remote redeployment in case the Payara admin is served via HTTPS using a self-signed certificate, which is the case of the default dataverse container installation.

For more context see this discussion: https://dataverse.zulipchat.com/#narrow/stream/375812-containers/topic/faster.20redeploy

This nginx configuration provides an HTTP endpoint at port 4849, and proxies requests to the Payara admin console's HTTPS 4848 endpoint.

**Special notes for your reviewer**:

You need IntelliJ IDEA Ultimate Edition to have remote deployment available in the Payara plugin.

**Suggestions on how to test this**:

Build and start the Dataverse containers:

```
mvn -Pct clean package
mvn -Pct docker:run
```

1. Go to the Payara admin console (either at https://localhost:4848 or http://localhost:4849) and uninstall the dataverse application under `Applications`.
2. Install Payara plugin (https://plugins.jetbrains.com/plugin/15114-payara-platform-tools). You seem to need IDEA Ultimate Edition. You may apply for a free license here: https://www.jetbrains.com/community/education/
3. Add a remote run config and configure it to connect to localhost 4849 and then run it.
4. Try `Run > Debugging actions > Update '...' application (cmd+f10) > Redeploy`
5. In ~20 seconds it should rebuild and redeploy the application inside docker
